packages/cli/commands/upload.js

```diff
@@ -237,6 +247,46 @@ exports.handler = async options => {
   }
 };
 
+/*
+ * Walks the src folder for files, filters them based on ignore filter.
+ * If processFieldsJs is true then will check for any JS fields conflicts (i.e., JS fields file and fields.json file) and prompt to resolve
+ */
+const getUploadableFileList = async (src, processFieldsJs) => {
```

Comment:
This was done so that we do not need to require the `cmsFieldPrompt` inside of `cli-lib`. This code was previously spread across `uploadFolder()` and `getFilesByType()` inside of `cli-lib/lib/uploadFolder.js`.

This is a *bit* slower now, because if `processFieldsJs` is true, we end up looping over the `filePaths` twice (once in this function and once in `getFilesByType()`) but it's very minor all things considered.

Reply:
What is the reason that we need to loop over things twice? Is it possible to do it once and pass that data to both places that need it?

Reply:
It is possible: we could just pull the entire `getFilesByType` function into this file and pass back the tuple to uploadFolder. I don't have a strong opinion one way or the other, so I will do it that way instead.